src/Symfony/Component/Ldap/Adapter/ExtLdap/Collection.php

```diff
@@ -44,31 +44,40 @@ public function toArray()
 
     public function count()
     {
-        if (false !== $count = ldap_count_entries($this->connection->getResource(), $this->search->getResource())) {
-            return $count;
+        $con = $this->connection->getResource();
+        $searches = $this->search->getResources();
+        $count = 0;
+        foreach ($searches as $search) {
```

Comment:
Can we keep the old way of counting without pagination to avoid n requests?

Reply:
Hmm... sure, but currently without pagination you should only have the single set of results and thus one trip to through the loop and one request. I hadn't done anything special previously because I figured the PHP overhead was fairly minimal.